src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -511,11 +511,44 @@ private BlittableJsonReaderObject RecreateCountersIfNeeded(DocumentsOperationCon
                 {
                     _documentsStorage.Delete(context, lowerId, id, null, lastModifiedTicks, changeVector, collectionName,
                         nonPersistentFlags, flags);
-                    return;
+                    return null;
                 }
+
+                document = ChangeSnapshotFlag(context, document, id, Constants.Documents.Metadata.RevisionCounters, Constants.Documents.Metadata.Counters);
+
                 _documentsStorage.Put(context, id, null, document, lastModifiedTicks, changeVector,
                     flags, nonPersistentFlags);
             }
+
+            return document;
+        }
+
+        private static BlittableJsonReaderObject ChangeSnapshotFlag(DocumentsOperationContext context, BlittableJsonReaderObject document, string documentId, string snapshotFlag, string flag)
+        {
+            if (document.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject metadata) &&
+                metadata.TryGet(snapshotFlag, out BlittableJsonReaderObject bjro))
+            {
+                bool isSnapshot = snapshotFlag == Constants.Documents.Metadata.RevisionCounters;
+
+                var names = bjro.GetPropertyNames();
```

Comment:
`bjro` is `@counters-snapshot` value, right? Maybe give it more meaningful name?

Reply:
done (we took this method from v5.2, and there it's also used for TS snapshot)

Reply:
okay, I didn't notice it's more generic usage here. it can stay as is